controllers/cassandrarestore_controller.go

```diff
@@ -66,10 +66,12 @@ func (r *CassandraRestoreReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 
 	restore := instance.DeepCopy()
 
-	if len(restore.Status.RestoreKey) == 0 {
+	if len(restore.Status.RestoreKey) != 0 {
```

Comment:
How come you flipped the logic here? We will requeue indefinitely with this change.

Reply:
Fixed